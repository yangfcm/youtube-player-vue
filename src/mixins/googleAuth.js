import { mapActions } from "vuex";
const googleAuth = {
  created() {
    if (!window.gapi) return;
    window.gapi.load("client:auth2", () => {
      window.gapi.client
        .init({
          clientId: process.env.VUE_APP_CLIENT_ID,
          scope:
            "email https://www.googleapis.com/auth/userinfo.profile https://www.googleapis.com/auth/youtube https://www.googleapis.com/auth/youtube.force-ssl",
        })
        .then(() => {
          this.auth = window.gapi.auth2.getAuthInstance();
          this.onAuthChange(this.auth.isSignedIn.get());
          this.auth.isSignedIn.listen(this.onAuthChange);
        });
    });
  },

  methods: {
    ...mapActions({
      signIn: "signIn",
      signOut: "signOut",
    }),
    onAuthChange(isSignedIn) {
      if (isSignedIn) {
        this.signIn(this.auth.currentUser.get()); // signIn action
      } else {
        this.signOut(); // signOut action
      }
    },
    handleGoogleSignout() {
      this.auth.signOut(); // signOut provided by Google Auth API
    },
    handleGoogleSignin() {
      this.auth.signIn(); // signIn provided by Google Auth API
    },
  },
};

export default googleAuth;
