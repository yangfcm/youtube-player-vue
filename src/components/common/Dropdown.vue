<template>
  <div class="ui vertical menu">
    <router-link class="item" to="/">Recommend</router-link>
    <router-link class="item" to="/subscription">Subscriptions</router-link>
    <router-link class="item" to="/playlist">Play list</router-link>
    <a class="item app-signout-btn" @click="handleSignout">Sign out</a>
  </div>
</template>

<script>
import googleAuth from "../../mixins/googleAuth";
export default {
  methods: {
    handleSignout() {
      this.handleGoogleSignout();
    }
  },
  mixins: [googleAuth]
};
</script>

<style scoped>
</style>