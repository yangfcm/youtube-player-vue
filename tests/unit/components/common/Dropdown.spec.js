import { shallowMount, RouterLinkStub } from "@vue/test-utils";
import Dropdown from "@/components/common/Dropdown.vue";
import googleAuth from "@/mixins/googleAuth";

jest.mock("@/mixins/googleAuth");

describe("Test Dropdown component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(Dropdown, {
      stubs: {
        RouterLink: RouterLinkStub,
      },
    });
  });
  afterEach(() => {
    wrapper.destroy();
  });

  it("Dropdown component matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Click sign out button can call handleSignout", () => {
    wrapper.find("a.app-signout-btn").trigger("click");
    expect(googleAuth.methods.handleGoogleSignout).toHaveBeenCalled();
  });
});
