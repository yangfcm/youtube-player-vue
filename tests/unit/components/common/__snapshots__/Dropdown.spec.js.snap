// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Dropdown component Dropdown component matches snapshot 1`] = `<div class="ui vertical menu"><a class="item">Recommend</a> <a class="item">Subscriptions</a> <a class="item">Play list</a> <a class="item app-signout-btn">Sign out</a></div>`;
